common/src/main/java/com/wynntils/models/beacons/BeaconModel.java

```diff
@@ -6,26 +6,42 @@
 
 import com.wynntils.core.WynntilsMod;
 import com.wynntils.core.components.Model;
+import com.wynntils.core.text.StyledText;
 import com.wynntils.mc.event.RemoveEntitiesEvent;
 import com.wynntils.mc.event.SetEntityDataEvent;
 import com.wynntils.mc.event.TeleportEntityEvent;
 import com.wynntils.models.beacons.event.BeaconEvent;
+import com.wynntils.models.beacons.event.BeaconMarkerEvent;
 import com.wynntils.models.beacons.type.Beacon;
-import com.wynntils.models.beacons.type.LootrunBeaconKind;
+import com.wynntils.models.beacons.type.BeaconKind;
+import com.wynntils.models.beacons.type.BeaconMarker;
+import com.wynntils.models.beacons.type.BeaconMarkerKind;
+import com.wynntils.utils.colors.CommonColors;
 import com.wynntils.utils.mc.McUtils;
 import it.unimi.dsi.fastutil.ints.Int2ObjectArrayMap;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import net.minecraft.core.component.DataComponents;
 import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.entity.Display;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.alchemy.PotionContents;
+import net.minecraft.world.item.component.CustomModelData;
 import net.neoforged.bus.api.EventPriority;
 import net.neoforged.bus.api.SubscribeEvent;
 
 public class BeaconModel extends Model {
+    public static final int COLOR_CUSTOM_MODEL_DATA = 79;
```

Comment:
Where is this int exactly? Is it not in the MC source code as a constant?

Reply:
It was in the original `LootrunBeaconKind` as that would do the check for unknwon beacon kind but now it's done in the model